From a4df0f455801bfc7a812cef9f3a996ce16e02535 Mon Sep 17 00:00:00 2001
From: krp97 <k.plata@metrological.com>
Date: Wed, 1 Sep 2021 13:45:37 +0000
Subject: [PATCH 6/6] Configure sinks to use the async mode

This commit configures brcmaudiosink and brcmvideosink to use the async
mode during state transition. During a READY -> PAUSED state change,
the nexus handles are not available, which results in a playback
in the paused state, because the system clock cannot be frozen.

Upon enabling the async state change, the overriden preroll function
will retry to freeze the system clock.
---
 reference/audiosink/src/gst_brcm_audio_sink.c | 4 ++--
 reference/videosink/src/gst_brcm_video_sink.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/reference/audiosink/src/gst_brcm_audio_sink.c b/reference/audiosink/src/gst_brcm_audio_sink.c
index 83f80be..60a25dc 100644
--- a/reference/audiosink/src/gst_brcm_audio_sink.c
+++ b/reference/audiosink/src/gst_brcm_audio_sink.c
@@ -51,7 +51,7 @@
 #include "bstd.h"
 #include "bmedia_types.h"
 
-#define OVERRIDE_PREROLL (0)
+#define OVERRIDE_PREROLL (1)
 #define ENABLE_UNUSED_FUNCTIONS (0)
 
 GST_DEBUG_CATEGORY_STATIC (brcmaudiosink);
@@ -314,7 +314,7 @@ static void gst_brcm_audio_sink_init (Gstbrcmaudiosink *sink) {
     gst_pad_set_unlink_function (GST_BASE_SINK_PAD (sink), GST_DEBUG_FUNCPTR (gst_brcm_audio_sink_sink_unlink));
 
     gst_base_sink_set_sync (GST_BASE_SINK (sink), FALSE);
-    gst_base_sink_set_async_enabled (GST_BASE_SINK (sink), FALSE);
+    gst_base_sink_set_async_enabled (GST_BASE_SINK (sink), TRUE);
 
     /* for bluetooth */
     char * audioDelay = getenv ("GST_AUDIO_OUTPUT_DELAY_MS");
diff --git a/reference/videosink/src/gst_brcm_video_sink.c b/reference/videosink/src/gst_brcm_video_sink.c
index 66f9196..a9b289a 100755
--- a/reference/videosink/src/gst_brcm_video_sink.c
+++ b/reference/videosink/src/gst_brcm_video_sink.c
@@ -304,7 +304,7 @@ static void gst_brcm_video_sink_init (Gstbrcmvideosink *sink) {
     } else {
         gst_base_sink_set_sync (GST_BASE_SINK (sink), FALSE);
     }
-//    gst_base_sink_set_async_enabled (GST_BASE_SINK (sink), FALSE);
+    gst_base_sink_set_async_enabled (GST_BASE_SINK (sink), TRUE);
 
     sink->upstream_peer_pad = NULL;
 
-- 
2.7.4

